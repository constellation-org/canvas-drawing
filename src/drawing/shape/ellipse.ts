import Shape, { ShapeToDraw } from './shape'

type EllipseToDraw = Omit<ShapeToDraw, 'path'> & {
  x: number,
  y: number,
  width: number,
  height: number
}

export default class Ellipse extends Shape {
  constructor(args: EllipseToDraw) {
    let path = new Path2D()
    path.ellipse(
      (args.x + args.width / 2),
      (args.y + args.height / 2),
      args.width / 2,
      args.height / 2,
      Math.PI,
      0,
      2 * Math.PI
    )
    super({...args, path })
  }
  draw(context: CanvasRenderingContext2D): void {
    super.draw(context) 
  }
}
