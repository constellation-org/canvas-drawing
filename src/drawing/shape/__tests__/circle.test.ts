import { drawEllipse } from '../ellipse'

describe('[Ellipse]', () => {
  let canvas: HTMLCanvasElement
  let context: CanvasRenderingContext2D
  beforeEach(function () {
    canvas = document.createElement('canvas')
    canvas.setAttribute('width', "100")
    canvas.setAttribute('height', "100")
    context = canvas.getContext('2d') as CanvasRenderingContext2D
  })
  it('should draw circle on top left corner', () => {
    drawEllipse(context, { x: 0, y: 0, width: 20, height: 20})
    expect(canvas.toDataURL('image/png')).toMatchSnapshot()
  })
  it('should draw circle on bottom right corner', () => {
    drawEllipse(context, { x: 80, y: 80, width: 20, height: 20})
    expect(canvas.toDataURL('image/png')).toMatchSnapshot()
  })
  it('should draw ellipse in the middle', () => {
    drawEllipse(context, { x: 30, y: 40, width: 40, height: 20})
    expect(canvas.toDataURL('image/png')).toMatchSnapshot()
  })
})
