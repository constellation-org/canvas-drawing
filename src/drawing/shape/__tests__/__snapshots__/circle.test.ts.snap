// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[Ellipse] should draw circle on bottom right corner 1`] = `"data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAGQAAABkCAYAAABw4pVUAAAAAXNSR0IArs4c6QAAA0BJREFUeF7tmMvrTV0cxj8MGDAwchkRZSYleZGk3HKJgQF6vQOlMMLIv0ApZmRAuYSUUhRCcntLUgYGBsTEZUYxMaFV+9Rq93POPs5PPYPPrj05+9nf8/R8znNWa03AKyqBCVFuNINAwn4EAhFIWAJhdmyIQMISCLNjQwQSlkCYHRsikLAEwuzYEIGEJRBmx4YIJCyBMDs2RCBhCYTZsSECCUsgzI4NEUhYAmF2bIhAwhIIs2NDBBKWQJgdGyKQsATC7NgQgYQlEGbHhggkLIEwOzZEIGEJhNmxIQIJSyDMjg0RSFgCYXZsiEDCEgizY0MEEpZAmB0bIpCwBMLs2BCBhCUQZseGCCQsgTA7NkQgYQmE2bEhAglLIMyODRFIWAJhdmyIQMISCLNjQwQSlkCYHRsikLAEwuzYEIGEJRBmx4YIJCyBMDs2RCBhCYTZsSECCUsgzI4NEUhYAmF2bIhAwhIIs2NDBBKWQJgdGyKQsATC7NgQgYQlEGbHhggkLIEwOzZEIGEJ/D07m4F/gEXN/RN40dzPgBtjfbUNGX8gc4HzwPIBo58C/wFva51AxhfIHuA4MLXj2G/AAeBMTy+Qjsl1kO0DTnbQjSXZD5wqDwTyhwm2XpsHvASmtD6/AFwHHjefrwC2Artauu/AQuCNQEYHMhF41FozPgH/Avd/M35ts87MqJ4/AVYKZHQgS4H/W2NKE0rA/a7VwN2WYJlARgdysFnIe5NOA3s7ji2L+e5Ke0ggHZPrI7sMbK+erwfudBy7qbUfuSKQjsn1kb0DZlfPpwFfO46dBXyotO8F0jG5PrKygNeL80zgc8exRfux0n4WSMfk+shuAeVvqndtAMpnXa6NwM1KeFsgXWLrrzkCHK4k14BtHce2YR4VSMfk+sh2AJdaz9cA9waM3tJsGmvZToGMDqTszss+ZEE1qqwr5eCwvc/oSdY1G8Pp1TuvgCUCGR1ImVBgPAcmtcZdBK42wCYDi5u/s7KLr68fzRH9K4GMD5Aypb1BHGbyIeBEeUEgw8Q2WFt26MeGOH7/0hy/n+uNFsjgkIdVzAHOAqsGvPigOYCsN4Y2ZNi0h9DPL6e31V1+/A+r+/VYs34BG2FMIdXJyIYAAAAASUVORK5CYII="`;

exports[`[Ellipse] should draw circle on top left corner 1`] = `"data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAGQAAABkCAYAAABw4pVUAAAAAXNSR0IArs4c6QAAA3RJREFUeF7tmcurTlEYxn/HgAEDJi4josykJCFJySWKgQHCQCmMMOJPoBQzMiAkpJSQS4rIJUkZGBgQE5cZhYEBemvvWi3f/r519lbnqe9ZtSdnv+/az/k959lrr3VGaB4bgYXAgur6A7ysrufAjT69vtWSwEiPvtnAeWDpgDmfADuAdy2f7bYeBHJDdgHHgEmFtL4D+4DThfUuG0AgNWQPcKIlsb3AyZa9bksI1IbMAV4BEzM6Z4HrwCNgHLAMiLVle1b3A5gPvDXdbgTCkAAdwNM142MF/X7D9KuqdWZacv8xsBz43U3ScHeHIYuBpxmGSEIA7jdWAveygiXAs+FG2u23D0P2Vwt5PdMpYHfhtLGY70xqDwDHC3td1vCVdQnYnNxbA9wtpLU+249cBrYU9rqswZD3wMzk3mTgWyGtGUCsN/X4AMwq7HVZgyGfgXRxngJ8LaQ1HfiU1H4B4mceLQnEGnIbiNdUPdYCdwrnWwfcTGqjL/o9WhIIQw4DB5P+q8CmwvlyM48Ahwp7XdbwyopF+GJ2L/7ybw0gtgG4ltVsBeIjwaMlgUhI7M5jHzIvWwtiN57vM+qS1dXGcGrS8xpYBPxsqcVtQH10Ema8AMZnVC4AVyrDJlTH8fE625bV/aqO6MMUjw4E0sPFfIM4mmm9IRwNrT61+fF77NCPjuL4PT6P4/j93H/SM/TT9PoHVWzszgArBtB5UL260o3h0APtCqCXIfWcc6vT2zjBjStqHybXm64Pd/+/BPoZYl5jQMCGjAH0fo+0ITZEjICYHCfEhogREJPjhNgQMQJicpwQGyJGQEyOE2JDxAiIyXFCbIgYATE5TogNESMgJscJsSFiBMTkOCE2RIyAmBwnxIaIERCT44TYEDECYnKcEBsiRkBMjhNiQ8QIiMlxQmyIGAExOU6IDREjICbHCbEhYgTE5DghNkSMgJgcJ8SGiBEQk+OE2BAxAmJynBAbIkZATI4TYkPECIjJcUJsiBgBMTlOiA0RIyAmxwmxIWIExOQ4ITZEjICYHCfEhogREJPjhNgQMQJicpwQGyJGQEyOE2JDxAiIyXFCbIgYATE5TogNESMgJscJsSFiBMTkOCE2RIyAmBwnxIaIERCT44TYEDECYnKcEBsiRkBMjhNiQ8QIiMlxQmyIGAExOU6IDREjICbnL8X+UWWCSl7XAAAAAElFTkSuQmCC"`;

exports[`[Ellipse] should draw ellipse in the middle 1`] = `"data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAGQAAABkCAYAAABw4pVUAAAAAXNSR0IArs4c6QAAA7pJREFUeF7tnMvrTGEcxj+/hYSNkEtuRZS13JKNS5R7yIIspMTGn+AfsLFy2whl567IbUNhoawUUW65x4pE0jfv6DTNTPM7Z2bep87z1mlmMed9n/f5nOe9nZohXKQcGJJSYzEYiNhDYCAGIuaAmBwnxEDEHBCT44QYiJgDYnKcEAMRc0BMjhNiIGIOiMlxQgxEzAExOU6IgYg5ICbHCTEQMQfE5DghBiLmgJgcJ8RAxBwQk+OEGIiYA2JynBADEXNATI4TYiBiDojJcUIMRMwBMTlOiIGIOSAmxwkxEDEHxOQ4IQYi5oCYHCfEQMQcEJPjhBhIKQdGAUuBucBsYBYwBRibrkmp1g/At3S9A56n6xlwD/hRqvUB3qSckMnAJmAdsBwIKFVKwLgNXAEuAO+rVNave9WAhJ6VwF5gAzCiTx3/BVwEjiRIfWpm+NUqAVkEHAbic5DlQXoAHg+y0XZtKQCZCBwCdkLHf5b4DtwF3gCv0+cr4C3wMnVwJjAVmAFMA6anK+af0R0M/wMcBw4CH3OCyQ1kTBo6VrQxISbpy2nMv1rRqLVpToqhMB6CVuUGsB74WbGt0rfnBhLjeBjUXE4Ap1IiSneuw43LgF3Anha/OQds6Uej3dSZE0isoM43iYzlaRh1vxvxPfjN4gR+TlNdm1Mqe9DE8KrICeQasLogNybXMChHibYXFhq+DqzJISQnkNgHNDZ00fcYRmLSzlFiDrtZaDjmrtgHDbwoAYnN352BO/CvwWj7Vt2BKA1ZD4EFdR+yFCb1GCZPprOxYjhrOamHAXGmtLHFMBXL3qPAoz4NYfOB/cDuFvXHUjweliwl5xwSHR6ZNn6r2vT+CXAGOJ1251VMil18nAbsAOa1qShgbK/zxrAB5SwQw0SnEsfnsRz9DHwBPgFf0/c4SomjkfHpGgdMSN/jM5bXcXzSqQSMrcDvKtSr3ps7IUX9OQ8XDwCxF8lelIA0zIgl6L40t/Tz+P0ScAyI8yuZogikYU5szLYBcShY3NGXNc8vqMo61+K+OBVekl7hxuvbeI3b7SvcF8BTv8LtIY06VaU8ZNWJw/++GogYdgMxEDEHxOQ4IQYi5oCYHCfEQMQcEJPjhBiImANicpwQAxFzQEyOE2IgYg6IyXFCDETMATE5ToiBiDkgJscJMRAxB8TkOCEGIuaAmBwnxEDEHBCT44QYiJgDYnKcEAMRc0BMjhNiIGIOiMlxQgxEzAExOU6IgYg5ICbHCTEQMQfE5DghBiLmgJgcJ0QMyF8jBXNlHzN2aAAAAABJRU5ErkJggg=="`;
